enhancements/ingress/wildcard-admission-policy.md

```diff
@@ -0,0 +1,194 @@
+
+OpenShift version 3 supports this feature through the `ROUTER_ALLOW_WILDCARD_ROUTES` environment variable when
+configuring the OpenShift Router directly. Users of this feature are blocked from upgrading to OpenShift 4 unless they
+discontinue use of the feature.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+See the [Wildcard Admission Policy section](#wildcard-admission-policy)
+
+## Motivation
+
+See the [Wildcard Admission Policy section](#wildcard-admission-policy)
+
+### Goals
+
+1. Define an API field that allows a cluster administrator to configure an ingress controller for admitting routes based
+on a route's wildcard policy.
+2. Provide similar wildcard route support as OpenShift version 3.
+
+### Non-Goals
+
+1. Redesign how wildcard support is implemented in [OpenShift Router](https://github.com/openshift/router).
+2. Introduce a separate API from `RouteAdmissionPolicy` for managing route admission based on a route's wildcard
+policy.
+
+## Proposal
+
+Introduce a `WildcardPolicy` field to the [IngressController](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)
+resource that allows administrators to manage route admission based on the route's wildcard policy configuration.
+
+### User Stories
+
+As a cluster administrator, I need the ability to admit routes based on the route's wildcard policy configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) will use `WildcardPolicy` to configure
+the environment variable `ROUTER_ALLOW_WILDCARD_ROUTES` in the [OpenShift Router](https://github.com/openshift/router).
+
+The default behavior of an ingress controller is to admit routes with a wildcard policy of None, which is backwards
+compatible with existing `IngressController` resources.
+
+### Risks and Mitigations
+
+Addition of the new field requires no mitigation, as the default is not changing.
+
+## Design Details
+
+### New API Field
+
+`WildcardPolicy` enum definition.
+
+```go
+// (Existing fields omitted, only newly proposed fields are visible)
+type RouteAdmissionPolicy struct {
+	// wildcardPolicy describes how routes with wildcard policies should
+	// be handled for the ingress controller. WildcardPolicy controls use
+	// of routes [1] exposed by the ingress controller based on the route's
+	// wildcard policy.
+	//
+	// [1] https://github.com/openshift/api/blob/master/route/v1/types.go
+	//
+	// If WildcardPolicy is updated from WildcardsAllowed to WildcardsDisallowed,
+	// routes admitted by the ingress controller with a wildcard policy of
+	// Subdomain must be updated to a wildcard policy of None to be readmitted.
```

Comment:
Based on your testing, would an even stronger warning be appropriate? My understanding is the newly rejected routes will _immediately stop receiving traffic_ and must be recreated or updated to force a re-admission. All this amounts to disruption of workloads, which is potentially catastrophic.

Reply:
@ironcladlou I updated the docs to draw greater attention to this caveat.